### ISSUE TITLE
How to improve Confidence for YOLOv8?

### ISSUE THREAD

**Issue by vinycecard** [ultralytics/ultralytics#15784]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am having some issues. I'm using the COCO 2017 dataset to train YOLOv8 models ('http://images.cocodataset.org/annotations/annotations_trainval2017.zip', 'http://images.cocodataset.org/zips/train2017.zip', 'http://images.cocodataset.org/zips/val2017.zip'). I am only using a few classes (like cars, bicycles, holes, and people). However, I am experiencing low confidence scores. Is there something we can do to increase the confidence? I ask because I see YOLO examples always with a bbox confidence above 0.8, and for some reason, I am getting less than that. Can you help me with this?

Another thing I wanted to check is if YOLO has any method for model conversion. The models are using very high weights (int32 or int64), and I would like something like (int16 or int8). Do you guys have any suggestions for achieving this? Thank you in advance for your response. Below, I detail how I am configuring the training model.
![image (3)](https://github.com/user-attachments/assets/ffb3bd59-a7bf-42c5-bce3-fa5883198fd0)
![fdf](https://github.com/user-attachments/assets/aed557c2-6101-465f-b3c8-2fea8bf20f4d)

#Code
results = model.train(
    data=configFilePath,
    epochs=150,
    imgsz=640,
    batch=50,
    optimizer='AdamW',
    augment=True,
    single_cls=True,
    cos_lr=True,
    lr0=1e-3,
    lrf=1e-2,
    momentum=0.95,
    weight_decay=2.5e-4,
    warmup_epochs=5.0,
    warmup_momentum=0.8,
    warmup_bias_lr=0.75,
    box=5.0,
    cls=0.75,
    dfl=2.0,
    pose=10.0,
    kobj=0.1,
    dropout=0.5,
    hsv_h=0.25,
    hsv_s=0.75,
    hsv_v=0.5,
    mosaic=1.0,
    mixup=0.1,
    copy_paste=0.1,
    erasing=0.5,
    crop_fraction=1.0,
    freeze=22
)

### Additional

Thanks in advance

**Comment by Skillnoob** [comment]
Do you have @Y-T-G blocked or something like that? He can't reply.

His reply:
You set the learning rate very low and you are using `freeze = 22` which freezes most of the model and only the last layer is trained, so it can't learn much.
You should only specify the arguments that are needed like `data`, `epochs` and `batch` and leave the rest as default.

**Comment by vinycecard** [comment]
Actually, even without freeze I was getting these lower rates. Right before I set these arguments, I was getting these lower resolts and I do not know why. Some of these arguments I am using is due to other ultralytics forums that learns how to for some specofic stuff I am working on...

**Comment by SmartestWashingMachine** [comment]
Maybe increase the `cls` weight value while training? That places a greater emphasis on classifying the bounding box class, which usually leads to larger probability values in inference. The exact ideal value will obviously depend on your dataset, but in my experiments I've found increasing it to `2` changes the average confidence score from `0.5` to `0.8+`

**Comment by glenn-jocher** [comment]
Thank you for your suggestion. Increasing the `cls` weight value during training can indeed help improve confidence scores. Adjusting it to `2` is a good starting point, but you may need to experiment with different values to find the optimal setting for your specific dataset.

**Comment by Y-T-G** [comment]
What do your labels look like? Did you label all the potholes?
